airflow/configuration.py

```diff
@@ -608,6 +609,48 @@ def getjson(self, section, key, fallback=_UNSET, **kwargs) -> Union[dict, list,
         except JSONDecodeError as e:
             raise AirflowConfigException(f'Unable to parse [{section}] {key!r} as valid json') from e
 
+    def gettimedelta(self, section, key, fallback=None, **kwargs) -> Optional[datetime.timedelta]:
+        """
+        Gets the config value for the given section and key, and converts it into datetime.timedelta object.
+        If the key is missing, then it is considered as `None`.
+
+        :param section: the section from the config
+        :param key: the key defined in the given section
+        :param fallback: fallback value when no config value is given, defaults to None
+        :raises AirflowConfigException: raised because ValueError or OverflowError
+        :return: datetime.timedelta(seconds=<config_value>) or None
+        """
+        val = self.get(section, key, fallback=fallback, **kwargs)
+
+        if val:
+            # the given value must be convertible to integer
+            try:
+                int_val = int(val)
+            except ValueError:
+                raise AirflowConfigException(
+                    f'Failed to convert value to int. Please check "{key}" key in "{section}" section. '
+                    f'Current value: "{val}".'
+                )
+
+            # the given value must be positive.
+            if int_val <= 0:
+                raise AirflowConfigException(
+                    f'Failed to convert value to timedelta in `seconds`. '
+                    f'Value must be greater than zero. '
+                    f'Please check "{key}" key in "{section}" section. Current value: "{val}".'
+                )
```

Comment:
Why? timedelta support negative values just fine.

Reply:
I thought that in the case of non-positive values, tasks reach the timeout immediately and this did not make sense to me.  So, restricted non-positive values. However, I guess I see your point, users can be held responsible for adjusting this parameter accordingly, then we don't need to handle this in the method, right?

Reply:
If you want to add a non-negative check, do it outside of `gettimedelta`. I'm fine with having the check, but it's done at the wrong place right now.

Reply:
I've removed the integer value check from gettimedelta method. Thank you for your comment @uranusjr, could you please review it again?